content/features/proxy2/install.md

```diff
@@ -0,0 +1,108 @@
++++
+title = 'Install the built-in proxy'
+weight = 20
++++
+
+The Memcached built-in proxy is available only if you activate a specific configuration option while compiling Memcached. For this reason, you can run the proxy only if you build Memcached from source. This page guides you through through the following tasks:
+
+* Build and install Memcached with its proxy features enabled.
+* Download the Lua library that the proxy requires for common tasks.
+
+For a general overview of the proxy, see [Proxy]({{<legacy_proxy_base_path>}}). For a hands-on tutorial that guides you through compiling and running the proxy with a simple configuration, see [Built-in Proxy Quickstart]({{<proxy_base_path>}}quickstart).
+
+## Before you begin {#prereqs}
+
+Make sure that your machine has the necessary development tools and libraries
+to build Memcached from source. These requirements are probably available from your operating system's package manager.
+
+For example, on Debian, you can satisfy these requirements by having the following packages installed:
+
+* `gcc`
+* `make`
+* `libevent-dev`
+
+The steps on this page guide you though launching three Memcached servers on TCP ports `11211`, `11212`, and `11213`. If you already have Memcached services running on those ports and you don't want to disrupt them, then you should take either of these additional steps:
+
+* Substitute different port numbers for this tutorial.
+* Run this tutorial as written on a different machine that isn't running Memcached.
+
+## Download and build the proxy {#download}
+
+To download and compile Memcached with its proxy features enabled, follow these steps:
+
+1. Download the latest Memcached release tarball:
+
+    ```console
+    wget https://memcached.org/latest
+    ```
+
+1. Unzip and untar the tarball, then change your working directory to the source directory:
+
+    ```console
+    tar xzvf memcached-{{<var>}}VERSION{{</var>}}.tar.gz
+
+    cd memcached-{{<var>}}VERSION{{</var>}}
+    ```
+    
+    Replace <var>`VERSION`</var> with the version number in the tarball that you downloaded in the previous step.
+    
+1. Configure the Memcached build process to enable the built-in proxy:
+
+    ```console
+    ./configure --with-proxy {{<var>}}OTHER_FLAGS{{</var>}}
+    ```
+    
+    Replace <var>OTHER_FLAGS</var> with a list of any other other Memcached configuration options that you wish to enable. For a complete list of these options, run the command `./configure --help`.
+    
+    If you want to install the proxy-enabled Memcached on a file system that already has Memcached installed on it, and you want to continue using that Memcached separately from the proxy, then you can add flags to `./configure` to adjust the new binary's name or installed location. For example, the following command configures the build process to make a proxy-enabled Memcached binary named `memcached-proxy`, instead of `memcached`:
```

Comment:
I'm pretty sure this will confuse people.

The "proxy enabled memcached" is 100% identical to a normal memcached if you don't supply the proxy-related startup options. There's no practical reason to need to keep the binaries separate. A huge part of the convenience of the feature is that it can be both in the same binary.

But on reading this people might think if they build the proxy it'll somehow contaminate the binary.

Reply:
I think the typical idea here is you can give it a different _installation prefix_, ie: `./configure --enable-proxy --prefix=/usr/local` which will install into `/usr/local/bin` if the user already has a /usr/bin/memcached from the system. This is standard practice if you're on a distro and want a newer binary for some reason.

Reply:
Got it! I've removed all of this suggestion, from this page and others. I wrote extra-cautiously here, but you're the best judge of your users' comfort level and I'm happy to keep things simpler.

Reply:
Got it! I've removed all of this suggestion, from this page and others. I wrote extra-cautiously here, but you're the best judge of your users' comfort level and I'm happy to keep things simpler.

Reply:
In my experience the fewer "branches" for people to go down the better.